pkg/objstore/s3/s3.go

```diff
@@ -47,46 +48,64 @@ type Bucket struct {
 type Config struct {
 	Bucket       string
 	Endpoint     string
-	AccessKey    string
-	SecretKey    string
+	Credentials  *credentials.Credentials
 	Insecure     bool
 	SignatureV2  bool
 	SSEEnprytion bool
 }
 
 // RegisterS3Params registers the s3 flags and returns an initialized Config struct.
 func RegisterS3Params(cmd *kingpin.CmdClause) *Config {
-	var s3config Config
+
+	var (
+		s3Config  Config
+		accessKey string
+		secretKey string
+	)
 
 	cmd.Flag("s3.bucket", "S3-Compatible API bucket name for stored blocks.").
-		PlaceHolder("<bucket>").Envar("S3_BUCKET").StringVar(&s3config.Bucket)
+		PlaceHolder("<bucket>").Envar("S3_BUCKET").StringVar(&s3Config.Bucket)
 
 	cmd.Flag("s3.endpoint", "S3-Compatible API endpoint for stored blocks.").
-		PlaceHolder("<api-url>").Envar("S3_ENDPOINT").StringVar(&s3config.Endpoint)
+		PlaceHolder("<api-url>").Envar("S3_ENDPOINT").StringVar(&s3Config.Endpoint)
 
 	cmd.Flag("s3.access-key", "Access key for an S3-Compatible API.").
-		PlaceHolder("<key>").Envar("S3_ACCESS_KEY").StringVar(&s3config.AccessKey)
+		PlaceHolder("<access-key>").StringVar(&accessKey)
 
-	s3config.SecretKey = os.Getenv("S3_SECRET_KEY")
+	cmd.Flag("s3.secret-key", "Secrety key for an S3-Compatible API.").
+		PlaceHolder("<secret-key>").StringVar(&secretKey)
 
 	cmd.Flag("s3.insecure", "Whether to use an insecure connection with an S3-Compatible API.").
-		Default("false").Envar("S3_INSECURE").BoolVar(&s3config.Insecure)
+		Default("false").Envar("S3_INSECURE").BoolVar(&s3Config.Insecure)
 
 	cmd.Flag("s3.signature-version2", "Whether to use S3 Signature Version 2; otherwise Signature Version 4 will be used.").
-		Default("false").Envar("S3_SIGNATURE_VERSION2").BoolVar(&s3config.SignatureV2)
+		Default("false").Envar("S3_SIGNATURE_VERSION2").BoolVar(&s3Config.SignatureV2)
 
 	cmd.Flag("s3.encrypt-sse", "Whether to use Server Side Encryption").
-		Default("false").Envar("S3_SSE_ENCRYPTION").BoolVar(&s3config.SSEEnprytion)
-
-	return &s3config
+		Default("false").Envar("S3_SSE_ENCRYPTION").BoolVar(&s3Config.SSEEnprytion)
```

Comment:
Could we fix the SSEEnprytion typo? (I assume that's what it is)

Reply:
Yes, Sir.

Reply:
Done.